File: app/src/main/java/ru/melodin/fast/model/ListItem.java
package ru.melodin.fast.model;

import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.graphics.drawable.Drawable;

import androidx.annotation.NonNull;

public class ListItem {

    @NonNull
    private String title = "";

    @NonNull
    private Drawable icon = new ColorDrawable(Color.TRANSPARENT);

    private int id = -1;
    private boolean visible = true;

    public ListItem(int id, @NonNull String title, @NonNull Drawable icon) {
        this.id = id;
        this.title = title;
        this.icon = icon;
    }

    public ListItem() {
    }

    public boolean isVisible() {
        return visible;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setVisible(boolean visible) {
        this.visible = visible;
    }

    @NonNull
    public String getTitle() {
        return title;
    }

    public void setTitle(@NonNull String title) {
        this.title = title;
    }

    @NonNull
    public Drawable getIcon() {
        return icon;
    }

    public void setIcon(@NonNull Drawable icon) {
        this.icon = icon;
    }
}

File: app/src/main/java/ru/melodin/fast/model/ListItem.kt
package ru.melodin.fast.model

import android.graphics.drawable.ColorDrawable
import android.graphics.drawable.Drawable

open class ListItem(var type: Int, var id: Int, var title: String, var icon: Drawable) : Model() {

    var isVisible = true

    constructor(title: String, icon: Drawable) : this(TYPE_ITEM, -1, title, icon)

    constructor(id: Int, title: String, icon: Drawable) : this(TYPE_ITEM, id, title, icon)

    constructor(type: Int) : this(
        type, -1, "", ColorDrawable()
    )

    constructor(divider: Boolean) : this(
        if (divider) TYPE_DIVIDER else TYPE_ITEM,
        -1,
        "",
        ColorDrawable()
    )

    companion object {
        const val TYPE_ITEM = 0
        const val TYPE_DIVIDER = 1
        const val TYPE_SHADOW_PADDING = 2
    }
}
